src/overlap_check/ProgressBar.cpp

```diff
@@ -0,0 +1,68 @@
+
+
+#include "ProgressBar.hpp"
+
+#include <sstream>
+#include <iostream>
+#include <iomanip>
+
+#if defined(__unix__) || defined(__unix) || (defined(__APPLE__) && defined(__MACH__))
+#include <unistd.h>
+#endif
+
+#define BAR_WIDTH 72
+
+ProgressBar::ProgressBar() {
+  // initialize bar
+  set_value(0.0);
+}
```

Comment:
Constructor is not needed due to initialization occurring via class member initializers, so these 4 lines can be removed along with the declaration in the header file.